### PR TITLE
Add ISupportSampling and implement it on Telemetry types that support it.

### PR DESCRIPTION
**Pull request by upendras**

### CODE REVIEW

**Comment by upendras** [comment]
@SergeyKanzhelev, @vitalyf007, @JakubOleksy please have a look.

File: src/Core/Managed/Shared/Extensibility/Implementation/Telemetry.cs [hunk 1]
```diff
@@ -9,6 +9,14 @@ internal static class Telemetry
         public static void WriteEnvelopeProperties(this ITelemetry telemetry, IJsonWriter json)
         {
             json.WriteProperty("time", telemetry.Timestamp);
+
+            var samplingSupportingTelemetry = telemetry as ISupportSampling;
+
+            if ((samplingSupportingTelemetry != null) && (samplingSupportingTelemetry.SamplingPercentage < 100.0 - 1.0E-12))
```
Comment (SergeyKanzhelev) [inline src/Core/Managed/Shared/Extensibility/Implementation/Telemetry.cs]: - check that it is more than 0, discard if it less than zero

File: src/Core/Managed/Shared/Constants.cs [hunk 2]
```diff
@@ -18,5 +18,10 @@ internal class Constants
         internal const string EventSourceGroupTraitKey = "ETW_GROUP";
 
         internal const int MaxExceptionCountToSave = 10;
+
+        /// <summary>		
+        /// Sampling percentage to use if none is specified.		
+        /// </summary>		
+        internal const double DefaultSamplingPercentage = 100.0;
```
Comment (SergeyKanzhelev) [inline src/Core/Managed/Shared/Constants.cs]: other constants doesn't have comment. Can omit it here as well.

**Comment by SergeyKanzhelev** [comment]
:shipit: with the change of condition in serialization

**Comment by vitalyf007** [comment]
Looks good to me.

**Comment by JakubOleksy** [comment]
:shipit:

**Pull request merged**